### PR TITLE
feat: graphql space configuration plugin

### PR DESCRIPTION
**Pull request by carlbrugger**
## Please explain how to summarize this PR for the Changelog:
Introducing the GraphQL to Flatfile Blueprint plugin. This plugin can be used either to generate Flatfile Blueprint from GraphQL or to configure a Flatfile Space from Graphql. Several options are available to provide the GraphQL to the plugin. It either can be provided from a GraphQL API endpoint, a GraphQL schema file, or a custom callback function.

## Tell code reviewer how and what to test:
Getting started instructions available on readme.

## Summary by CodeRabbit

- **New Features**
    - Introduced the `@flatfile/plugin-graphql-schema` plugin, facilitating conversion of GraphQL schemas to Flatfile Blueprints for data validation and preparation.
    - Added functionality to generate Flatfile properties, sheets, and workbooks from GraphQL schemas, with customization options for schema generation.
- **Documentation**
    - Updated documentation to reflect the new plugin capabilities and usage guidelines.
- **Refactor**
    - Renamed the package from `@flatfile/plugin-graphql` to `@flatfile/plugin-graphql-schema` and updated the version to `1.0.0`.

### CODE REVIEW

File: plugins/graphql-schema/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@flatfile/plugin-graphql-schema",
+  "version": "0.0.0",
```
Comment (bangarang) [inline plugins/graphql-schema/package.json:3]:
```suggestion
  "version": "1.0.0",
```

File: plugins/graphql-schema/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# @flatfile/plugin-graphql
+# @flatfile/plugin-graphql-schema
 
-## 0.0.2
+## 0.0.0
```
Comment (bangarang) [inline plugins/graphql-schema/CHANGELOG.md:3]:
```suggestion
## 1.0.0
```

File: plugins/graphql-schema/src/setup.factory.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    )
+    return {
+      workbooks,
+      space: setupFactory.space,
+    }
+  } catch (error) {
+    console.error('Error generating workbook:', error)
+    throw new Error('Error generating workbook')
+  }
+}
```
[the remark this answers is not in the document]
Comment (carlbrugger) [reply plugins/graphql-schema/src/setup.factory.ts:24]: The `error` is being console logged. Don't want it in the the new Error being thrown as that message will be used in the job failure displayed in the UI.

File: plugins/graphql-schema/src/generate.field.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+function getScalarType(scalar) {
+  switch (scalar) {
+    case 'Float':
+    case 'Int':
+      return 'number'
+    case 'Boolean':
+      return 'boolean'
+    default:
+      return 'string'
+  }
```
[the remark this answers is not in the document]
Comment (carlbrugger) [reply plugins/graphql-schema/src/generate.field.ts:44]: This case is covered with `default`

**Review by bangarang — APPROVED** [review]

**Pull request merged**